tests/cecil-tests/AttributeTest.cs

```diff
@@ -107,35 +124,359 @@ public void DoubleAttributedElements (string assemblyPath)
+							var mentionedPlatforms = GetAvailabilityAttributes (member).ToList();
+							if (mentionedPlatforms.Any()) {
+								var claimedPlatforms = GetSupportedAvailabilityAttributes (member).ToList();
+								string key = GetMemberLookupKey (member);
+								if (!harvestedInfo.ContainsKey (key)) {
+									harvestedInfo[key] = new Dictionary<string, PlatformClaimInfo>();
+								}
+								var claimInfo = new PlatformClaimInfo (mentionedPlatforms, claimedPlatforms);
+								if (harvestedInfo[key].ContainsKey(currentPlatform)) {
+									harvestedInfo[key][currentPlatform].UnionWith (claimInfo);
+								}
+								else {
+									harvestedInfo[key][currentPlatform] = claimInfo;
+								}
+							}
+						}
+					}
+				}
+			}
+					
+			// Now walk every item found above and check two things:
+			var attributesWereCompiledOut = new List<string>();			
+			var doesNotExistWhereClaimed = new List<string>();
+			foreach (var (member, info) in harvestedInfo) {
+				// 1. All platforms match in count of mentioned (we did not conditionally compile out attributes)
+				int expectedPlatformCount = info.First().Value.MentionedPlatforms.Count();
+				if (info.Any (i => i.Value.MentionedPlatforms.Count() != expectedPlatformCount)) {
+						if (IgnoreElementsThatDoNotExistInThatAssembly (member)) {
+							continue;
+						}
+						string detailedPlatformBreakdown = string.Join ("\n", info.Select(x => ($"Assembly {x.Key} => {x.Value}")));
+						string errorMessage = $"{member} did not have the same number of SupportedOSPlatformAttribute in every assembly:\n{detailedPlatformBreakdown}";
+						attributesWereCompiledOut.Add (errorMessage);
+#if DEBUG
+						Console.Error.WriteLine (errorMessage);
+#endif
+				}
+
+
+				// 2. For each supported attribute claim exist, that it exists on that platform
+				// Since we know each platform claims are now equal, just use the first one
+				var claimedPlatforms = info.First().Value.ClaimedPlatforms;
+				foreach (var platform in claimedPlatforms) {
+					if (!info.ContainsKey (platform)) {
+						if (IgnoreElementsThatDoNotExistInThatAssembly (member)) {
+							continue;
+						}
+						string detailedPlatformBreakdown = string.Join ("\n", info.Select(x => ($"Assembly {x.Key} => Declares ({string.Join (" ", x.Value)})")));
+						string errorMessage = $"{member} was not found on {platform} despite being declared supported there.";
+						doesNotExistWhereClaimed.Add (errorMessage);
+#if DEBUG
+						Console.Error.WriteLine (errorMessage);
+#endif		
+					}
+				}
+			}
+
+			Assert.That (attributesWereCompiledOut, Is.Empty, $"{attributesWereCompiledOut.Count} issues found");
+			Assert.That (doesNotExistWhereClaimed, Is.Empty, $"{doesNotExistWhereClaimed.Count} issues found");
+		}
+
+		static bool IgnoreElementsThatDoNotExistInThatAssembly (string member)
+		{
+			// Xkit has many platform specific bits in AppKit/UIKit and is a mess to get right
+			if (member.StartsWith ("Kit")) {
```

Comment:
All this can be a lot nicer with a switch and since we have the new c# we could put all in the same switch.